pkgs/development/libraries/glfw/3.x.nix

```
@@ -17,17 +19,29 @@ stdenv.mkDerivation rec {
   propagatedBuildInputs = [ libGL ];
 
   nativeBuildInputs = [ cmake ]
-    ++ lib.optional stdenv.isDarwin fixDarwinDylibNames;
+    ++ lib.optional stdenv.isDarwin fixDarwinDylibNames
+    ++ lib.optional waylandSupport extra-cmake-modules;
 
-  buildInputs = [ libX11 libXrandr libXinerama libXcursor libXi libXext ]
-    ++ lib.optionals stdenv.isDarwin [ Cocoa Kernel ];
+  buildInputs =
+    if waylandSupport
+    then [ wayland wayland-protocols libxkbcommon ]
+    else [ libX11 libXrandr libXinerama libXcursor libXi libXext ]
+         ++ lib.optionals stdenv.isDarwin [ Cocoa Kernel ];
 
   cmakeFlags = [
     "-DBUILD_SHARED_LIBS=ON"
   ] ++ lib.optional (!stdenv.isDarwin) [
     "-DCMAKE_C_FLAGS=-D_GLFW_GLX_LIBRARY='\"${lib.getLib libGL}/lib/libGL.so.1\"'"
+  ] ++ lib.optional waylandSupport [
+    "-DGLFW_USE_WAYLAND=ON"
+    "-DCMAKE_C_FLAGS=-D_GLFW_EGL_LIBRARY='\"${lib.getLib libGL}/lib/libEGL.so.1\"'"
   ];
 
+  postPatch = lib.optional waylandSupport ''
+    substituteInPlace src/wl_init.c \
+        --replace "libxkbcommon.so.0" "${lib.getLib libxkbcommon}/lib/libxkbcommon.so.0"
```

Fix doubled indentation
```suggestion
      --replace "libxkbcommon.so.0" "${lib.getLib libxkbcommon}/lib/libxkbcommon.so.0"
```

---

Pure shell files use 4 spaces, nix files use 2 spaces. 3 spaces is not used anywhere.

---

Up to you. But it’s not a shell file, it’s a Nix file. For me it’s a lot more convenient if the same file follows the same indentation rules for all lines. Even for such pieces of code written in other languages. If following specific indentation rules that are different from local file is a necessity it’s better to store this chunk of code in a separate file if you ask me.

---

No, it is not up to you. nix files are using 2 spaces to intend and nothing else.

---

No, it is not up to you. nix files are using 2 spaces to intend and nothing else.

---

That's not true. We use 2 spaces to indent nix code, but nix doesn't execute what is in those bash scripts, it feeds it to bash, so it's not nix code, it's bash code in a nix string.
I think the indentation used in code embedded in an other language is always debatable. In litterate programming, org-mode, jupyter, html/css/js, etc. the indentation isn't consistent across a file.
Nix generates files, so should we prioritize the indentation in the nix file or the generated file? With python for example, I don't think it will be happy with a generated file with 2 spaces for indentation in 4-spaces-indented code.

Anyways the indentation/formatting is inconsistent in this repo, so it shouldn't matter.